DS-Session4-Question2/src/meta/test/uniCharTest.java

```diff
@@ -0,0 +1,49 @@
+package meta.test;
+
+import static org.junit.Assert.*;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import meta.utility.UniqueChar;
+
+@RunWith(Parameterized.class)
+public class uniCharTest {
+	static UniqueChar unique;
+	@BeforeClass
+	public static void start(){
+		unique = new UniqueChar();
+	}
+	@Parameters
+	public static Collection<Object[]> data(){
```

Comment:
As per question, "Since the counting operation can be time consuming, the methods should cache the results, so that when the methods is given a string previously encountered, it will simply retrieve the stored result."
This part is not being tested in the test cases.